lib/supavisor/db_handler.ex

```diff
@@ -379,4 +414,14 @@ defmodule Supavisor.DbHandler do
       auth.secrets.().user
     end
   end
+
+  @spec receive_ready_for_query() :: :ok | :timeout_error
+  defp receive_ready_for_query() do
+    receive do
+      {_proto, _socket, <<?Z, 5::32, ?I>>} ->
```

Comment:
I wonder if it is guaranteed that it will always be an individual message and if part of the TCP stack may concatenate this with other messages. I would definitely add logging to the `after` branch (although it will crash anyway, so maybe it is fine).

Reply:
after this error message, the pg backend only sends a `ready_for_query` message